app/src/main/java/org/fossasia/openevent/general/event/EventService.kt

```diff
@@ -63,20 +63,12 @@ class EventService(private val eventApi: EventApi, private val eventDao: EventDa
         }
     }
 
-    fun getSimilarEvents(id: Long): Flowable<List<Event>> {
-        val eventsFlowable = eventDao.getAllSimilarEvents(id)
-        return eventsFlowable.switchMap {
-            if (it.isNotEmpty())
-                eventsFlowable
-            else
-                eventTopicApi.getEventsUnderTopicId(id)
-                        .toFlowable()
-                        .map {
-                            eventDao.insertEvents(it)
-                        }
-                        .flatMap {
-                            eventsFlowable
-                        }
+    fun getSimilarEvents(id: Long, savedLocation: String): Single<List<Event>> {
+        return eventTopicApi.getEventsUnderTopicId(id, "name", savedLocation).flatMap { apiList ->
+            val eventIds = apiList.map { it.id }.toList()
+            eventDao.getFavoriteEventWithinIds(eventIds).flatMap { favIds ->
+                updateFavorites(apiList, favIds)
+            }
```

Comment:
@iamareebjamal The process is in an infinite loop. The same process occurs correctly for loading of events by location in main screen, but here even after the mapping is done correctly, the loading still occurs.